src/commit.c

```diff
@@ -518,3 +518,54 @@ int git_commit_nth_gen_ancestor(
 	*ancestor = parent;
 	return 0;
 }
+
+int git_commit_header_field(git_buf *out, const git_commit *commit, const char *field)
+{
+	const char *buf = commit->raw_header;
+	const char *h, *eol;
+
+	git_buf_sanitize(out);
+	while ((h = strchr(buf, '\n')) && h[1] != '\0' && h[1] != '\n') {
+		h++;
+		if (git__prefixcmp(h, field)) {
+			buf = h;
+			continue;
+		}
+
+		h += strlen(field);
+		eol = strchr(h, '\n');
+		if (h[0] != ' ' || !eol)
```

Comment:
Should this check actually be `if (h[0] != ' ') continue; if (!eol) goto malformed;`?

It looks like if you are asking for the `par` field, then we will match the prefix and then choke because it's not followed by a space...